src/scenes/Activities/scenes/EventHistory.js

```diff
@@ -48,134 +38,23 @@ const messages = defineMessages({ // eslint-disable-line
-  state = {
-    transactions: [],
-    order: SortBy.DESCENDING.toLowerCase(),
-    orderBy: 'createdTime',
-    perPage: 10,
-    page: 0,
-    limit: 50,
-    skip: 0,
-    expanded: [],
-  };
-
-  navigating = false
-
   componentDidMount() {
-    this.props.store.ui.location = Routes.ACTIVITY_HISTORY;
-    this.executeTxsRequest();
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { transactions, syncBlockNum } = this.props;
-
-    // Update page on new block
-    if (nextProps.syncBlockNum !== syncBlockNum) {
-      this.executeTxsRequest();
-    }
-
-    if (nextProps.transactions || transactions) {
-      const sorted = _.orderBy(
-        nextProps.transactions ? nextProps.transactions : transactions,
-        [this.state.orderBy],
-        [this.state.order],
-      );
-
-      this.setState({
-        transactions: sorted,
-      });
-    }
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    const { skip } = this.state;
-    const { oracles, history } = this.props;
-
-    if (skip !== prevState.skip) {
-      this.executeTxsRequest();
-    }
-
-    if (prevProps.oracles !== oracles && this.navigating) {
-      const path = getDetailPagePath(oracles);
-      if (path) history.push(path);
-    }
+    this.props.store.activities.activityHistory.init();
   }
 
   render() {
-    const { classes } = this.props;
-    const { transactions } = this.state;
-
-    return (
-      <Grid container spacing={0}>
-        {
-          transactions.length ?
-            (<Table className={classes.historyTable}>
-              {this.getTableHeader()}
-              {this.getTableRows()}
-              {this.getTableFooter()}
-            </Table>) :
-            (<Typography variant="body1">
-              <FormattedMessage id="str.emptyTxHistory" defaultMessage="You do not have any transactions right now." />
-            </Typography>)
-        }
-      </Grid>
-    );
-  }
-
-  executeTxsRequest = () => {
-    const { orderBy, order, limit, skip } = this.state;
-    const direction = order === SortBy.DESCENDING.toLowerCase() ? SortBy.DESCENDING : SortBy.ASCENDING;
-
-    this.props.getTransactions(
-      [
-        { type: TransactionType.APPROVE_CREATE_EVENT },
-        { type: TransactionType.APPROVE_CREATE_EVENT },
-        { type: TransactionType.BET },
-        { type: TransactionType.APPROVE_SET_RESULT },
-        { type: TransactionType.SET_RESULT },
-        { type: TransactionType.APPROVE_VOTE },
-        { type: TransactionType.VOTE },
-        { type: TransactionType.FINALIZE_RESULT },
-        { type: TransactionType.WITHDRAW },
-        { type: TransactionType.WITHDRAW_ESCROW },
-        { type: TransactionType.RESET_APPROVE },
-      ],
-      { field: orderBy, direction },
-      limit,
-      skip,
-    );
-  }
-
-  getTableHeader = () => {
+    const { classes, history } = this.props;
+    const { transactions, order, orderBy, page, perPage, pageChange, perPageChange } = this.props.store.activities.activityHistory;
```

Comment:
`const { transactions, order, orderBy, page, perPage, pageChange: onPageChange, perPageChange }  = this.props.store.activities.activityHistory;`